crates/cheatcodes/src/inspector.rs

```diff
@@ -1374,7 +1409,13 @@ impl Cheatcodes {
 
                     let zk_tx = if self.use_zk_vm {
                         // We shouldn't need factory_deps for CALLs
-                        Some(ZkTransactionMetadata { factory_deps: Default::default() })
+                        if call.target_address == DEFAULT_CREATE2_DEPLOYER_ZKSYNC {
+                            Some(ZkTransactionMetadata {
+                                factory_deps: create2_factory_deps.clone(),
```

Comment:
here. I think only the current contract is present, so it won't work for a factory
`fetch_all_factory_deps` takes care of getting also the contract's own, so no need to push it separately

Reply:
Actually the factory_dep code can be moved to here.
```rs
let zk_tx = if self.use_zk_vm {
    // We need factory deps for CREATE2, which is a CALL in zksync 
    if call.target_address == DEFAULT_CREATE2_DEPLOYER_ZKSYNC {
        // get factory_deps...
    } else {
        Some(ZkTransactionMetadata { factory_deps: Default::default() })
    }
} else {
    None
};
let factory_deps = zk_tx.clone().map(|tx| tx.factory_deps).unwrap_or_default();
```